scipy/sparse/sparsetools/sparsetools.cxx

```diff
@@ -280,7 +281,7 @@ call_thunk(char ret_spec, const char *spec, thunk_t *thunk, PyObject *args)
                 goto fail;
             }
 
-            if (PyArray_EquivTypenums(I_typenum, NPY_INT64)
+            if ((*p == 'l' || PyArray_EquivTypenums(I_typenum, NPY_INT64))
                     && value == (npy_int64)value) {
                 arg_list[j] = std::malloc(sizeof(npy_int64));
                 *(npy_int64*)arg_list[j] = (npy_int64)value;
```

Comment:
The `else if` clause following this should start with `else if (*p == 'i' && PyArray_EquivTypenums(I_typenum, NPY_INT32) ...`